tools/src/dump-cluster-spec/dump-cluster-spec.ts

```diff
@@ -33,7 +33,7 @@ async function main (opts: CommandOpts): Promise<void> {
 
   const cluster_spec = await client.get('/_plugins/api')
 
-  write_yaml(opts.output, cluster_spec.data)
+  write_yaml(opts.output, Buffer.from(cluster_spec.data as string, 'binary').toString())
```

Comment:
This needs to parse from JSON as well before writing. Probably a good idea to extract this `Buffer.from(` to a helper function as it's being used in a few places.

Reply:
I refactored things around to only need to use custom deserialization in the `ChapterReader` case.